test/python/transpiler/test_calibrationbuilder.py

```diff
@@ -233,76 +177,269 @@ def test_non_native_cr_with_target(self):
             builder.call_gate(SXGate(), qubits=(1,))
             builder.call_gate(RZGate(np.pi / 2), qubits=(1,))
 
-        self.assertEqual(schedule(test_qc, self.backend), target_qobj_transform(ref_sched))
+        return ref_sched
+
+    def build_stretched_pulse_forward(
+        self,
+        theta,
+        u0p_play,
+        d1p_play,
+    ):
+        """A helper function to generate reference pulse schedule for forward direction."""
+        duration = self.compute_stretch_duration(u0p_play, 2.0 * theta)
+        width = self.compute_stretch_width(u0p_play, 2.0 * theta)
+
+        with builder.build() as ref_sched:
+            # Positive CRs
+            u0p_params = u0p_play.pulse.parameters
+            u0p_params["duration"] = duration
+            u0p_params["width"] = width
+            builder.play(
+                GaussianSquare(**u0p_params),
+                ControlChannel(0),
+            )
+            d1p_params = d1p_play.pulse.parameters
+            d1p_params["duration"] = duration
+            d1p_params["width"] = width
+            builder.play(
+                GaussianSquare(**d1p_params),
+                DriveChannel(1),
+            )
+            builder.delay(duration, DriveChannel(0))
+
+        return ref_sched
+
+    def compute_stretch_duration(self, play_gaussian_square_pulse, theta):
+        """Compute duration of stretched Gaussian Square pulse."""
+        pulse = play_gaussian_square_pulse.pulse
+        sigma = pulse.sigma
+        width = self.compute_stretch_width(play_gaussian_square_pulse, theta)
+
+        duration = width + sigma * self.__risefall
+        return round(duration / self.__granularity) * self.__granularity
+
+    def compute_stretch_width(self, play_gaussian_square_pulse, theta):
+        """Compute width of stretched Gaussian Square pulse."""
+        pulse = play_gaussian_square_pulse.pulse
+        sigma = pulse.sigma
+        width = pulse.width
+
+        risefall_area = sigma * np.sqrt(2 * np.pi) * erf(self.__risefall)
+        full_area = risefall_area + width
+
+        target_area = abs(theta) / self.__angle * full_area
+        return max(0, target_area - risefall_area)
+
+    def u0p_play(self, cr_schedule):
+        """Returns the positive CR pulse from cr_schedule."""
+        return self.get_cr_play(cr_schedule, "CR90p_u")
+
+    def u0m_play(self, cr_schedule):
+        """Returns the negative CR pulse from cr_schedule."""
+        return self.get_cr_play(cr_schedule, "CR90m_u")
+
+    def d1p_play(self, cr_schedule):
+        """Returns the positive rotary echo pulse from cr_schedule."""
+        return self.get_cr_play(cr_schedule, "CR90p_d")
+
+    def d1m_play(self, cr_schedule):
+        """Returns the negative rotary echo pulse from cr_schedule."""
+        return self.get_cr_play(cr_schedule, "CR90m_d")
 
     def test_native_cr(self):
```

Comment:
Can you remove this test case from `TestCalibrationBuilder`? This must be a part of `TestRZXCalibrationBuilder` which is a subclass of `TestCalibrationBuilder`.

Reply:
Removed.